### PR TITLE
Awsviya 220

### PR DESCRIPTION
**Pull request by chlync**

### CODE REVIEW

File: templates/sas-viya-macro.template.yaml [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+#      KeyPairName:
+#        default: Key Name
+#      QSS3BucketName:
+#        default: Quick Start S3 Bucket Name
+#      QSS3KeyPrefix:
+#        default: Quick Start S3 Key Prefix
+#      DeploymentDataLocation:
+#        default: SAS Viya Software Order File
+#      CASInstanceCount:
+#        default: "The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy)"
```
Comment (sasgwb) [inline templates/sas-viya-macro.template.yaml]: There is a closing parenthesis here, but not an opening parenthesis

File: templates/sas-viya-master.template.yaml [hunk 2]
```diff
@@ -47,6 +48,8 @@ Metadata:
         default: Quick Start S3 Key Prefix
       DeploymentDataLocation:
         default: SAS Viya Software Order File
+      CASInstanceCount:
+        default: "The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy)"
```
Comment (enderm) [inline templates/sas-viya-master.template.yaml]: you might want to check with Penny, but I think CAS needs to be always all caps and
"singleton cas" should be "SMP CAS deployment"
"MPP deploy" should be "MPP CAS deployment"

File: templates/sas-viya-macro.template.yaml [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+#      - "r5a.24xlarge"
+#
+#    ConstraintDescription: Must contain either "i3" or "r5"
+#    Default: "i3.4xlarge"
+#    Description: Type and size of EC2 instance for the Viya Compute Node (choose i3 for performance or r5 for auto-recovery)
+#    Type: String
+#
+#  CASInstanceCount:
+#    Description: >-
+#      The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy).
```
Comment (sasgwb) [inline templates/sas-viya-macro.template.yaml]: See above comment.

File: templates/sas-viya-master.template.yaml [hunk 4]
```diff
@@ -124,6 +127,14 @@ Parameters:
     Description: Type and size of EC2 instance for the Viya Compute Node (choose i3 for performance or r5 for auto-recovery)
     Type: String
 
+  CASInstanceCount:
+    Description: >-
+      The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy).
```
Comment (enderm) [inline templates/sas-viya-master.template.yaml]: ibid

File: templates/sas-viya.template.yaml [hunk 5]
```diff
@@ -57,6 +58,8 @@ Metadata:
         default: SAS Viya Software Order File
       CASInstanceSize:
         default: EC2 Instance Size for the CAS compute VM
+      CASInstanceCount:
```
Comment (enderm) [inline templates/sas-viya.template.yaml]: ibid

File: templates/sas-viya.template.yaml [hunk 6]
```diff
@@ -57,6 +58,8 @@ Metadata:
...
+        default: "The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy)"
```
Comment (enderm) [inline templates/sas-viya.template.yaml]: ibid

File: templates/sas-viya-macro.template.yaml [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+      ^([-a-zA-Z0-9@:%_\+.~#?&//=]*)\.zip$
+    Description: >-
+      S3 location of the Software Order Confirmation e-mail attachment.
+      Example: mysasbucket/viya_deployment_data/SAS_Viya_deployment_data.zip
+    Type: String
+    MinLength: '1'
+    ConstraintDescription: must be a zip file in a valid S3 bucket location, e.g. mysasbucket/viya_deployment_data/SAS_Viya_deployment_data.zip
+  CASInstanceCount:
+    Description: >-
+      The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy).
```
Comment (sasgwb) [inline templates/sas-viya-macro.template.yaml]: See above comment.

File: templates/sas-viya.template.yaml [hunk 8]
```diff
@@ -91,6 +94,13 @@ Parameters:
     Type: String
     MinLength: '1'
     ConstraintDescription: must be a zip file in a valid S3 bucket location, e.g. mysasbucket/viya_deployment_data/SAS_Viya_deployment_data.zip
+  CASInstanceCount:
+    Description: >-
+      The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy).
```
Comment (enderm) [inline templates/sas-viya.template.yaml]: ibid

File: templates/sas-viya.template.yaml [hunk 9]
```diff
@@ -530,6 +540,9 @@ Conditions:
     Fn::Not:
     - Condition: NoSSLCertificate
 
+Transform:
+  - MacroTest4
```
Comment (enderm) [inline templates/sas-viya.template.yaml]: MacroTest4? not a production kinda name....

File: templates/sas-viya-master.template.yaml (hunk 2 above)
Comment (sasgwb) [inline templates/sas-viya-master.template.yaml]: Closing parenthesis but no opening parenthesis

File: templates/sas-viya-master.template.yaml (hunk 4 above)
Comment (sasgwb) [inline templates/sas-viya-master.template.yaml]: Closing parenthesis but no opening parenthesis

File: templates/sas-viya.template.yaml (hunk 6 above)
Comment (sasgwb) [inline templates/sas-viya.template.yaml]: Closing parenthesis but no opening parenthesis

File: templates/sas-viya.template.yaml (hunk 8 above)
Comment (sasgwb) [inline templates/sas-viya.template.yaml]: Closing parenthesis but no opening parenthesis

File: templates/subtemplates/sas-viya.casworker.template.yaml [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+    Default: CASController
+    Description: >-
+      The name of the case worker or controller to create.
+      Only change this value if you customize or extend the Quick Start for your own use.
+      This string can include numbers, lowercase letters, uppercase letters, and hyphens (-).
+      It cannot start or end with a hyphen (-).
+    Type: String
+  CASInstanceCount:
+    Description: >-
+      The number of Cas nodes to create. Note: selecting 1 will install a singleton cas, but selecting 2 or more will create an MPP deploy).
```
Comment (sasgwb) [inline templates/subtemplates/sas-viya.casworker.template.yaml]: Closing parenthesis but no opening parenthesis

File: templates/sas-viya-macro.template.yaml (hunk 1 above)
Comment (chlync) [reply templates/sas-viya-macro.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya-master.template.yaml (hunk 2 above)
Comment (chlync) [reply templates/sas-viya-master.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya-macro.template.yaml (hunk 3 above)
Comment (chlync) [reply templates/sas-viya-macro.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya-master.template.yaml (hunk 4 above)
Comment (chlync) [reply templates/sas-viya-master.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya.template.yaml (hunk 5 above)
Comment (chlync) [reply templates/sas-viya.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya.template.yaml (hunk 6 above)
Comment (chlync) [reply templates/sas-viya.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya-macro.template.yaml (hunk 7 above)
Comment (chlync) [reply templates/sas-viya-macro.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya.template.yaml (hunk 8 above)
Comment (chlync) [reply templates/sas-viya.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya.template.yaml (hunk 9 above)
Comment (chlync) [reply templates/sas-viya.template.yaml]: I think that is from a prior commit?

File: templates/sas-viya-master.template.yaml (hunk 2 above)
Comment (chlync) [reply templates/sas-viya-master.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya-master.template.yaml (hunk 4 above)
Comment (chlync) [reply templates/sas-viya-master.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya.template.yaml (hunk 6 above)
Comment (chlync) [reply templates/sas-viya.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/sas-viya.template.yaml (hunk 8 above)
Comment (chlync) [reply templates/sas-viya.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

File: templates/subtemplates/sas-viya.casworker.template.yaml (hunk 10 above)
Comment (chlync) [reply templates/subtemplates/sas-viya.casworker.template.yaml]: Updated, but Penny will also be reviewing and doing the final drafting of wording Thursday, so subject to change.

**Pull request merged**